tests/test_core.py

```diff
@@ -1061,6 +1061,134 @@ def test_select_P_ABBA_val_inf():
     npt.assert_almost_equal(ref, comp)
 
 
+def test_merge_topk_PI():
+    # `assume_unique = True`
+    n = 50
+    for k in range(1, 6):
+        PA = np.random.rand(n * k).reshape(n, k)
+        PA[:, :] = np.sort(PA, axis=1)  # sorting each row separately
+
+        PB = np.random.rand(n * k).reshape(n, k)
+        col_idx = np.random.randint(0, k, size=n)
+        for i in range(n):  # creating ties between values of PA and PB
+            PB[i, col_idx[i]] = np.random.choice(PA[i], size=1, replace=False)
+        PB[:, :] = np.sort(PB, axis=1)  # sorting each row separately
+
+        IA = np.arange(n * k).reshape(n, k)
+        IB = IA + n * k
+
+        ref_P = PA.copy()
+        ref_I = IA.copy()
+
+        comp_P = PA.copy()
+        comp_I = IA.copy()
+
+        naive.merge_topk_PI(ref_P, PB.copy(), ref_I, IB.copy())
+        core._merge_topk_PI(comp_P, PB.copy(), comp_I, IB.copy())
+
+        npt.assert_array_equal(ref_P, comp_P)
+        npt.assert_array_equal(ref_I, comp_I)
+
+    # `assume_unique = False`
+    n = 50
+    for k in range(1, 6):
+        PA = np.random.rand(n * k).reshape(n, k)
+        PB = np.random.rand(n * k).reshape(n, k)
+
+        IA = np.arange(n * k).reshape(n, k)
+        IB = IA + n * k
+
+        cols_idx_A = np.random.randint(0, k, size=n)
+        for i in range(n):
+            # create overlaps
+            IDX = np.random.choice(np.arange(k), cols_idx_A[i], replace=False)
+            PB[i, IDX] = PA[i, IDX]
+            IB[i, IDX] = IA[i, IDX]
```

Comment:
Is this a good way to create overlapping values? (see line 1102-1106)? 
Maybe I should do:
```
small_noise = (1 - np.random.rand(len(IDX)) * 2) * 1e-14
PB[i, IDX] = PA[i, IDX] + small_noise
IB[i, IDX] = IA[i, IDX]
```

Reply:
I don't see anything immediately wrong with what your `np.random.choice` approach. Would it be helpful to do:

```
IDX = np.random.choice(np.arange(k), cols_idx_A[i], replace=False)
imprecision = np.random.uniform(low=-1e6, high=1e6, size=len(IDX))
PB[i, IDX] = PA[i, IDX] + imprecision  
IB[i, IDX] = IA[i, IDX]
```

Here, I'm randomly adding some small imprecision (note "adding" can also mean subtracting)

Reply:
> Here, I'm randomly adding some small imprecision (note "adding" can also mean subtracting)

Got it. (btw, the name `imprecision` is definitely better than `noise`). I think I will add the `imprecision`. I think it  will create a more realistic input for  our tests. Thanks :)